app/javascript/components/FiltersBox.js

```diff
@@ -1,76 +1,165 @@
   //   * filter-species-NCBItaxon9606
   const facetName = props.facet.name;
+  const facetId = props.facet.id;
   const componentName = 'filters-box';
-  const filtersboxId = `${componentName}-${props.facet.id}`;
-  const saveID = `save-${filtersboxId}`;
+  const filtersBoxId = `${componentName}-${facetId}`;
+  const applyId = `apply-${filtersBoxId}`;
+
+  const filtersSearchBarId = `filters-search-bar-${filtersBoxId}`;
 
   /**
    * Returns IDs of selected filters.
-   * Enables comparing current vs. saved filters to enable/disable SAVE button
+   * Enables comparing current vs. applied filters to enable/disable APPLY button
    */
-  function getCheckedFilterIDs() {
-    const checkedSelector = `#${filtersboxId} input:checked`;
-    const checkedFilterIDs =
+  function getCheckedFilterIds() {
+    const checkedSelector = `#${filtersBoxId} input:checked`;
+    const checkedFilterIds =
       [...document.querySelectorAll(checkedSelector)].map((filter) => {
         return filter.id;
       });
-    return checkedFilterIDs
+    return checkedFilterIds
+  }
+
+  function updateSelections() {
+    const checkedFilterIds = getCheckedFilterIds();
+    setSelection(checkedFilterIds);
+    setShowClear(checkedFilterIds.length > 0);
+  }
+
+  function handleApplyClick(event) {
+    const applyButtonClasses = Array.from(event.target.classList);
+
+    if (applyButtonClasses.includes('disabled')) return;
+
+    const checkedFilterIds = getCheckedFilterIds();
+
+    if (checkedFilterIds.length > 0) {
+      searchContext.facets[facetId] = checkedFilterIds.join(',');
+    } else {
+      delete searchContext.facets[facetId];
+    }
+
+    setAppliedSelection(checkedFilterIds);
   }
 
-  function handleFilterClick() {
-    setSelection(getCheckedFilterIDs());
+  function clearFilters() {
+    const checkedSelector = `#${filtersBoxId} input:checked`;
+    document.querySelectorAll(checkedSelector).forEach((checkedInput) => {
+      checkedInput.checked = false;
+    });
+
+    updateSelections();
   }
 
-  function handleSaveClick(event) {
-    const saveButtonClasses = Array.from(event.target.classList);
+  // Search for filters in this facet that match input text terms
+  //
+  // For example, among the many filters in the "Disease" facet, search
+  // for filters matching the term "tuberculosis".
+  async function searchFilters(terms) {
+    const apiData = await fetchFacetFilters(props.facet.id, terms);
+    const matchingFilters = apiData.filters;
+
+    setHasFilterSearchResults(apiData.query !== '' && matchingFilters.length > 0);
 
-    if (saveButtonClasses.includes('disabled')) return;
+    setMatchingFilters(matchingFilters);
+  }
+
+  async function handleFilterSearchSubmit(event) {
+    event.preventDefault();
+    const terms = event.target.elements[filtersSearchBarId].value;
+    await searchFilters(terms);
+  }
 
-    setSavedSelection(getCheckedFilterIDs());
-  };
+  async function handleFilterSearchButtonClick(event) {
```

Comment:
on further reflection, how about the SearchBarFilter component does the event handling, and you just pass it a single performSearch(searchTerms) function that it can call whenever the button is pressed or enter is hit.  That way SearchFilterBar is handling it's own events and UI components, and this component just has to handle executing the search

Reply:
Done in 09c35ea.  Great suggestions!

I started off trying to handle events in `FiltersSearchBar`, but abandoned that approach after having trouble doing the search filters API call.  Your suggestion to pass `performSearch(searchTerms)` was key to refactoring.

I've retained my previous naming for that function, `searchFilters(terms)`, but am now passing that in as a prop into `FiltersSearchBar` as you recommend.

That, in turn, enabled me to handle `onSubmit` and `onClick` events on `FiltersSearchBar` in that component itself, as you suggested.  Cohesion is better that way.

And I've done away with the brittle DOM traversal, and use `getElementById` as you suggested.